### PR TITLE
feature/6182-ApptAnalytics

### PR DESCRIPTION
**Pull request by Sparowhawk**
## Description of Change
added appointment analytics and some generic analytics for components

## Screenshots/Video
N/A

## Testing
yarn test

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations

Existing event name | Type of update | Details
-- | -- | --
vama_ttv_appt_details | REMOVE | Current logic doesn't make sense and the concept can be measured other ways
vama_appt_cancel | Add new params | apt_idapt_statusapt_typedays_to_apt
vama_appt_view_details | Add new params | apt_idapt_statusapt_typedays_to_apt

New event name | event name length | Logic description | event_param(s)
-- | -- | -- | --
vama_segcontrol_click | 13 | Makes a selection from any instance of SegmentedControl | label
vama_modalpick_open |   | Clicks to open any instance of VAModalPicker | modal
vama_modalpick_sel |   | Makes a selection from any instance of VAModalPicker | modalselection
vama_accordion_click | 18 | Clicks to expand or close accordions | section isOpened
vama_apt_add_cal | 16 | Clicks Add to calendar | apt_idapt_statusapt_typedays_to_apt
vama_apt_cancel_clicks | 21 | Fires when user clicks in-app Cancel button to cancel an appointment or appointment request and when they click to confirm that first click | apt_idapt_statusapt_typedays_to_aptstep

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: VAMobile/src/screens/HealthScreen/Appointments/AppointmentDetailsCommon/PendingAppointmentCancelButton.tsx [hunk 1]
```diff
@@ -26,11 +29,26 @@ const PendingAppointmentCancelButton: FC<PendingAppointmentCancelButtonProps> =
...
+    if (attributes.status === AppointmentStatusConstants.CANCELLED) {
+      apiStatus = 'Canceled'
+    } else if (attributes.status === AppointmentStatusConstants.BOOKED) {
+      apiStatus = 'Confirmed'
+    } else if (isPendingAppointment) {
+      apiStatus = 'Pending'
+    }
+    const apptDate = Math.floor(DateTime.fromISO(attributes.startDateUtc).toMillis() / (1000 * 60 * 60 * 24))
+    const nowDate = Math.floor(DateTime.now().toMillis() / (1000 * 60 * 60 * 24))
+    const days = apptDate - nowDate
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/Appointments/AppointmentDetailsCommon/PendingAppointmentCancelButton.tsx:43]: This logic is used in a few files, can this be refactored into a function and stored in the appointments or analytics util file?

Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/Appointments/AppointmentDetailsCommon/PendingAppointmentCancelButton.tsx]: agreed, pulled out to separate functions for that logic

File: VAMobile/src/store/slices/appointmentsSlice.ts [hunk 2]
```diff
@@ -327,17 +327,17 @@ export const getAppointmentsInDateRange =
  * Redux action to cancel appointment associated with the given cancelID
  */
 export const cancelAppointment =
-  (cancelID?: string, appointmentID?: string, isPendingAppointment?: boolean): AppThunk =>
+  (cancelID?: string, appointmentID?: string, isPendingAppointment?: boolean, status?: string, type?: string, days_to_apt?: number): AppThunk =>
```
Comment (theodur) [inline VAMobile/src/store/slices/appointmentsSlice.ts:330]: Super Nit, but I noticed the last param `days_to_apt` isn't camel case in a few places in this file

**Review by theodur — APPROVED** [review]

**Comment by narin** [comment]
Conflict but otherwise LGTM

**Review by theodur — APPROVED** [review]
Since Dylan is out, I went ahead and resolved the merge conflict. This should be good now

**Pull request merged**